components/brave_wallet/browser/android_swap_page_appearing_browsertest.cc

```diff
@@ -0,0 +1,408 @@
+    mojo::PendingReceiver<cosmetic_filters::mojom::CosmeticFiltersResources>
+        receiver) {
+  g_brave_browser_process->ad_block_service()->GetTaskRunner()->PostTask(
+      FROM_HERE, base::BindOnce(&BindCosmeticFiltersResourcesOnTaskRunner,
+                                std::move(receiver)));
+}
+}  // namespace
+
+class TestWebUIControllerFactory : public content::WebUIControllerFactory {
+ public:
+  explicit TestWebUIControllerFactory(const std::string& source_name)
+      : source_name_(source_name) {}
+
+  std::unique_ptr<content::WebUIController> CreateWebUIControllerForURL(
+      content::WebUI* web_ui,
+      const GURL& url) override {
+    if (url.host_piece() == kWalletPageHost) {
+      return std::make_unique<SwapPageUI>(web_ui, source_name_);
+    }
+
+    return nullptr;
+  }
+
+  content::WebUI::TypeID GetWebUIType(content::BrowserContext* browser_context,
+                                      const GURL& url) override {
+    if (url.SchemeIs(content::kChromeUIScheme) &&
+        url.host_piece() == kWalletPageHost && url.path_piece() == "/swap") {
+      return reinterpret_cast<content::WebUI::TypeID>(1);
+    }
+
+    return content::WebUI::kNoWebUI;
+  }
+
+  bool UseWebUIForURL(content::BrowserContext* browser_context,
+                      const GURL& url) override {
+    return url.SchemeIs(content::kChromeUIScheme) || url == "about:blank";
+  }
+
+ private:
+  std::string source_name_;
+};
+
+class AndroidBrowserTestSwap : public PlatformBrowserTest {
+ public:
+  AndroidBrowserTestSwap() {
+    factory_ = std::make_unique<TestWebUIControllerFactory>(kWalletPageHost);
+    content::WebUIControllerFactory::RegisterFactory(factory_.get());
+    scoped_feature_list_.InitWithFeatures(
+        {}, {features::kBraveWalletFilecoinFeature,
+             features::kBraveWalletSolanaFeature});
+  }
+
+  void SetUpOnMainThread() override {
+    InitWallet();
+    SetEthChainIdInterceptor();
+    PlatformBrowserTest::SetUpOnMainThread();
+  }
+
+  content::WebContents* GetActiveWebContents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+  Profile* GetProfile() { return chrome_test_utils::GetProfile(this); }
+
+  PrefService* GetPrefs() {
+    return TabModelList::models()[0]->GetProfile()->GetPrefs();
+  }
+
+  base::FilePath get_temp_path() const { return temp_dir_.GetPath(); }
+  int64_t file_size() const { return file_size_; }
+  absl::optional<std::string> file_digest() const { return file_digest_; }
+
+  const std::string GetConsoleMessages(
+      const content::ConsoleObserver& console_observer) const {
+    std::stringstream ss;
+    for (auto const& msg : console_observer.messages()) {
+      ss << msg.message << " [" << msg.log_level << "]"
+         << "source_id:" << msg.source_id << " line_no:" << msg.line_no
+         << std::endl;
+    }
+    return ss.str();
+  }
+
+ protected:
+  class TestContentBrowserClient : public ChromeContentBrowserClient {
```

Comment:
why subclass `ChromeContentBrowserClient` and add Brave specific code manually instead of subclassing `BraveContentBrowserClient`? This does not seem like a good way to test this because the in `BraveContentBrowserClient` might change or not be the same

Reply:
I missed the existence of BraveContentBrowserClient. I'll fix it.